packages/sanity/src/structure/panes/documentList/useDocumentSheetColumns.tsx

```diff
@@ -44,43 +51,93 @@ const PreviewCell = (props: {
   )
 }
 
-const TableTextInput = (props: any) => {
-  const {index, id} = props
-  const initialValue = props.getValue()
-  // We need to keep and update the state of the cell normally
-  const [value, setValue] = useState(initialValue || '')
+const columnHelper = createColumnHelper<SanityDocument>()
+const SUPPORTED_FIELDS = ['string', 'number', 'boolean']
 
-  // When the input is blurred, we'll call our table meta's updateData function
-  const onBlur = () => {
-    props.table.options.meta?.updateData(index, id, value)
-  }
+type Columns = (
+  | AccessorKeyColumnDef<SanityDocument, unknown>
+  | GroupColumnDef<SanityDocument, unknown>
+)[]
 
-  const handleChange = useCallback(
-    (e: FormEvent<HTMLInputElement>) => setValue(e.currentTarget.value),
-    [],
-  )
+const getColsFromSchemaType = (schemaType: SchemaType, parentalField?: string): Columns => {
+  //@ts-expect-error - wip.
+  return schemaType.fields.reduce((cols: Columns[], field: any) => {
```

Comment:
I see this is from the old code but I think you can remove the `// @ts-expect-error` by using the `ObjectSchemaType` instead of `SchemaType` as the `ObjectSchemaType` is a more narrow type that has the `fields` accessible. Additionally `.reduce` takes a type parameter that you can use to specify the type of the accumulator. Specifying the type parameter there will enforce that your reducer returns the correct type while leaving the incoming `field` parameter typed with the type of the array item.

I also see that this code has `cols` typed as `Columns[]` instead of `Columns` (the `Columns` type has an array already attached to it). Was this intended?

```suggestion
const getColsFromSchemaType = (schemaType: ObjectSchemaType, parentalField?: string): Columns => {
  return schemaType.fields.reduce<Columns>((cols, field) => {
```

Reply:
- [ ] Remove WIP types for `getColumns`

Reply:
Thanks for that suggestion @ricokahler - I didn't know about the type property on `reduce`